Mf2/Parser.php

```diff
@@ -663,73 +682,96 @@ public function parseDT(\DOMElement $dt, &$dates = array()) {
 			foreach ($valueClassChildren as $e) {
 				if (strstr(' ' . $e->getAttribute('class') . ' ', ' value-title ')) {
 					$title = $e->getAttribute('title');
-					if (!empty($title))
+					if (!empty($title)) {
 						$dateParts[] = $title;
+					}
 				}
 				elseif ($e->tagName == 'img' or $e->tagName == 'area') {
 					// Use @alt
 					$alt = $e->getAttribute('alt');
-					if (!empty($alt))
+					if (!empty($alt)) {
 						$dateParts[] = $alt;
+					}
 				}
 				elseif ($e->tagName == 'data') {
 					// Use @value, otherwise innertext
 					$value = $e->hasAttribute('value') ? $e->getAttribute('value') : unicodeTrim($e->nodeValue);
-					if (!empty($value))
+					if (!empty($value)) {
 						$dateParts[] = $value;
+					}
 				}
 				elseif ($e->tagName == 'abbr') {
 					// Use @title, otherwise innertext
 					$title = $e->hasAttribute('title') ? $e->getAttribute('title') : unicodeTrim($e->nodeValue);
-					if (!empty($title))
+					if (!empty($title)) {
 						$dateParts[] = $title;
+					}
 				}
 				elseif ($e->tagName == 'del' or $e->tagName == 'ins' or $e->tagName == 'time') {
 					// Use @datetime if available, otherwise innertext
 					$dtAttr = ($e->hasAttribute('datetime')) ? $e->getAttribute('datetime') : unicodeTrim($e->nodeValue);
-					if (!empty($dtAttr))
+					if (!empty($dtAttr)) {
 						$dateParts[] = $dtAttr;
+					}
 				}
 				else {
-					if (!empty($e->nodeValue))
+					if (!empty($e->nodeValue)) {
 						$dateParts[] = unicodeTrim($e->nodeValue);
+					}
 				}
 			}
 
 			// Look through dateParts
 			$datePart = '';
 			$timePart = '';
+			$timezonePart = '';
 			foreach ($dateParts as $part) {
 				// Is this part a full ISO8601 datetime?
-				if (preg_match('/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}(?::\d{2})?(?:Z?[+|-]\d{2}:?\d{2})?$/', $part)) {
+				if (preg_match('/^\d{4}-\d{2}-\d{2}T?\d{2}:\d{2}(?::\d{2})?(?:Z?[+|-]\d{2}:?\d{2})?$/', $part)) {
 					// Break completely, we’ve got our value.
 					$dtValue = $part;
 					break;
 				} else {
 					// Is the current part a valid time(+TZ?) AND no other time representation has been found?
 					if ((preg_match('/\d{1,2}:\d{1,2}(Z?[+|-]\d{2}:?\d{2})?/', $part) or preg_match('/\d{1,2}[a|p]m/', $part)) and empty($timePart)) {
 						$timePart = $part;
+
+						$timezoneOffset = normalizeTimezoneOffset($timePart);
+						if (!$impliedTimezone && $timezoneOffset) {
+							$impliedTimezone = $timezoneOffset;
+						}
 					} elseif (preg_match('/\d{4}-\d{2}-\d{2}/', $part) and empty($datePart)) {
 						// Is the current part a valid date AND no other date representation has been found?
 						$datePart = $part;
+					} elseif (preg_match('/(Z?[+|-]\d{2}:?\d{2})/', $part) and empty($timezonePart)) {
```

Comment:
Since the time may also only be hours, updated this to `/^(Z|[+-]\d{2}:?(\d{2})?)$/`